### PR TITLE
Reorganize Docs Layout

### PR DESCRIPTION
**Pull request by derekpierre**
Based over #1728 .

Reorganize our doc's table of contents - hopefully, this makes the docs a bit easier to follow.
- Categorize guides by running a node vs application development
- Add a blurb to WorkLock Overview *(mostly taken from the Figment AMA)*
- Utilize rst reference linking where applicable to reduce URL issues when pages are moved/renamed
- Remove 'NuCypher' from page titles since it is redundant

Related to #1600 .

Screen Shot 2020-02-21 at 12 04 45 PM

Screen Shot 2020-02-21 at 12 05 05 PM

Screen Shot 2020-02-21 at 12 05 22 PM

### CODE REVIEW

File: docs/source/guides/worklock_guide.rst [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+`WorkLock` is a novel token distribution mechanism we created that requires participants to run NuCypher nodes
+to receive NU tokens. Think of it as a replacement for an ICO or an airdrop as a way to distribute initial
+tokens, but with a stronger filter for desirable participants (in our case, people who will stake and run nodes on
+the network).
+
+There will be an open contribution period in which anyone can contribute ETH to the WorkLock contract. That ETH will
+be escrowed on-chain and at the end of the contribution period, stake-locked NU will be distributed pro rata across
+contributors. If those contributors use that stake-locked NU to run a node, the NU will eventually unlock and their
+escrowed ETH will be returned in full. Alternatively, WorkLock participants can cancel their bid to forgo NU and
+recoup their escrowed ETH.
```
Comment (mswilkison) [inline docs/source/guides/worklock_guide.rst]:
```suggestion
recoup their escrowed ETH immediately.
```

File: docs/source/guides/worklock_guide.rst [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
 
 Overview
 --------
 
-TODO
+`WorkLock` is a novel, permissionless token distribution mechanism, developed at NuCypher, which requires participants to stake ETH and maintain NuCypher nodes in order to receive NU tokens.
+
+WorkLock offers specific advantages over ICO or airdrop as a distribution mechanism, chiefly: it selects for participants who are most likely to strengthen the network because they commit to staking and running nodes.
+
+The WorkLock begins with an open bidding period, during which anyone seeking to participate can send ETH to the WorkLock contract to be escrowed on-chain.  At the end of the bidding period, stake-locked NU will be distributed pro rata across contributors. If those contributors use that stake-locked NU to run a node, the NU will eventually unlock and their escrowed ETH will be returned in full. Alternatively, WorkLock participants can cancel their bid to forgo NU and recoup their escrowed ETH immediately.
```
Comment (arjunhassard) [inline docs/source/guides/worklock_guide.rst]:
```suggestion
The WorkLock begins with an open bidding period, during which anyone seeking to participate can send ETH to the WorkLock contract to be escrowed on-chain.  At the end of the bidding period, stake-locked NU will be distributed pro rata across participants. If those participants use that stake-locked NU to run a node and earn inflationary rewards, the NU will eventually unlock and their escrowed ETH will be returned in full. During the WorkLock, participants who change their mind are able to cancel their bid, forgoing the NU and recouping their escrowed ETH immediately.
```

File: docs/source/support/faq.rst [hunk 3]
```diff
@@ -2,6 +2,12 @@
 Frequently Asked Questions
 ==========================
 
+Is there a Wiki?
+----------------
+
+Yes - https://github.com/nucypher/wiki/wiki
```
Comment (arjunhassard) [inline docs/source/support/faq.rst]: This is currently a private repo but I think it needs a small update and to be opened to community contributions. It's better than non-NuCypherinos attempting to update RTD

File: docs/source/guides/development/development.rst [hunk 4]
```diff
@@ -0,0 +1,12 @@
+=======================
+Application Development
+=======================
+
+The NuCypher Network provides cryptographic infrastructure for privacy-preserving applications.
```
Comment (arjunhassard) [inline docs/source/guides/development/development.rst]: Looking forward to adding much more stuff here!!

File: docs/source/support/faq.rst (hunk 3 above)
Comment (derekpierre) [reply docs/source/support/faq.rst]: Added as part of #1728 but now deleted (https://github.com/nucypher/nucypher/pull/1728#discussion_r382870665). We can always add it once the relevant updates are made.

File: docs/source/guides/development/development.rst (hunk 4 above)
Comment (derekpierre) [reply docs/source/guides/development/development.rst]: 👍

File: docs/source/guides/worklock_guide.rst (hunk 2 above)
Comment (derekpierre) [reply docs/source/guides/worklock_guide.rst]: Just double-checking - any concern here about stating that cancelling can occur "**during**" worklock. Is it "during", or "after" (eg. a cancel phase)...or are they all one in the same anyway?

OR should we just keep it general that "WorkLock participants can cancel..." without specifying when? 
(@vzotova?)

**Pull request merged**